### PR TITLE
[Elastic Agent] Rename *ConfigChange to PolicyChange

### PR DESCRIPTION
**Pull request by ph**
This rename the ConfigChagne to PolicyChange for:

- The internal type for the actions.
- The handler
- The introspections

This depends on change on Kibana to return the new Type and the new
"Policy" key.

The kibana API need to be adjusted to send the right action type and the right fields.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Fixes: #20687 20687

### CODE REVIEW

**Review by michalpristas — APPROVED** [review]
looks ok, it either needs to be synced with fleet changes or made in a non-breaking two step process when we accept both types with same result and then we remove `CONFIG_CHANGE` later

**Comment by ph** [comment]
I've talked to @michalpristas and he will take it over from me.

**Comment by jen-huang** [comment]
Hi @michalpristas, I have Kibana changes here: https://github.com/elastic/kibana/pull/78221

The Kibana changes includes BWC for <= 7.9 agent. I tested changing policies with a 7.9.1 snapshot build of agent and it worked well. Could you verify BWC as well as with the agent changes in this PR? Thanks!

**Comment by michalpristas** [comment]
@jen-huang tested with agent from this branch. it retrieves action of type `POLICY_CHANGE` as it should.

File: x-pack/elastic-agent/CHANGELOG.asciidoc [hunk 1]
```diff
@@ -84,6 +84,18 @@
...
+- Remove support for logs type and use logfile {pull}19761[19761]
+- Avoid comparing uncomparable types on enroll {issue}19976[19976]
+- Fix issues with merging of elastic-agent.yml and fleet.yml {pull}20026[20026]
+- Unzip failures on Windows 8/Windows server 2012 {pull}20088[20088]
+- Fix failing unit tests on windows {pull}20127[20127]
+- Improve GRPC stop to be more relaxed {pull}20118[20118]
+- Prevent closing closed reader {pull}20214[20214]
+- Fix Windows service installation script {pull}20203[20203]
+- Fix timeout issue stopping service applications {pull}20256[20256]
+- Fix rename *ConfigChange to *PolicyChange to align on changes in the UI. {pull}20779[20779]
```
Comment (ph) [inline x-pack/elastic-agent/CHANGELOG.asciidoc:98]: oops changelog issue?

Comment (michalpristas) [reply x-pack/elastic-agent/CHANGELOG.asciidoc:98]: wou

Comment (michalpristas) [reply x-pack/elastic-agent/CHANGELOG.asciidoc]: changelog vs changelog.next

**Pull request merged**